pkgs/applications/networking/hyprspace/default.nix

```diff
@@ -0,0 +1,27 @@
+{ iproute2mac ? null, lib, stdenv, buildGoModule, fetchFromGitHub }:
+
+assert stdenv.isDarwin -> iproute2mac != null;
+
+buildGoModule rec {
+  pname = "hyprspace";
+  version = "0.1.2";
+
+  propagatedBuildInputs = lib.optional stdenv.isDarwin iproute2mac;
```

Comment:
you don't need propagatedBuildInputs here, buildInputs should suffice.
You can try to check the references to a derivation that you built (a closure) like so
`nix-store --query --references ./result/`
iproute2mac should appear in the final references.

Reply:
Would `iproute2mac` appear in $PATH with buildInputs (though i'm not sure if propagatedBuildInputs does that either)? From what i read in the README of hyprspace, it's needed at runtime. Alternatively we could create a wrapper script that puts iproute2mac in the $PATH.

Reply:
It depends, if there is an explicit reference, then it will appear.
I just think it needs to be tested once.
If it doesn't appear, then propagatedBuildInputs is the right choice.